### PR TITLE
Add command line argument properties to build scans

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

It has been pointed out that it may be easier to debug issues with gradle enterprise if the command line arguments were also published.

ref: https://ge.armeria.dev/s/nwvvmqptvhxro

Modifications:

- Add a whitelist set of arguments that may be published to our build scans
- Go through `rootProject.properties` and add arguments to the build scan

Result:

- We can easily debug build cache issues when going through build scans.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -350,3 +350,18 @@ class TestsReportTask extends DefaultTask {
         }
     }
 }
+
+// additional configuration that can't be done at settings.gradle
+gradleEnterprise {
+    buildScan {
+        // maintain a whitelist so that sensitive information (credentials) aren't accidentally published.
+        Set<String> whiteList = ['coverage', 'leak', 'blockhound', 'noLint',
```
Comment (ribafish) [inline build.gradle:358]: I also see `-Pretry=true` and `-PnoWeb` properties in github actions invocations in case they were missed 👍

**Review by minwoox — APPROVED** [review]
👍 👍 👍

File: build.gradle [hunk 2]
```diff
@@ -430,3 +430,19 @@ configure(relocatedProjects) {
         }
     }
 }
+
+// additional configuration that can't be done at settings.gradle
+gradleEnterprise {
+    buildScan {
+        // maintain a whitelist so that sensitive information (credentials) aren't accidentally published.
+        Set<String> whiteList = ['coverage', 'leak', 'blockhound', 'noLint', 'flakyTests', 'buildJdkVersion',
```
Comment (trustin) [inline build.gradle:438]: whitelist/whiteList -> allowlist/allowList

**Review by ikhoon — APPROVED** [review]
👍👍

**Pull request merged**